WickedEngine/wiECS.h

```diff
@@ -346,6 +406,103 @@ namespace wi::ecs
 		// Disallow this to be copied by mistake
 		ComponentManager(const ComponentManager&) = delete;
 	};
+
+	// This is the class to store all ComponentManagers,
+	// this is useful for bulk operation of all attached components within an entity
+	class ComponentLibrary
+	{
+	public:
+		wi::unordered_map<uint32_t, std::unique_ptr<ComponentManager_Interface>> componentManagers;
+
+		// Create an instance of ComponentManager of a certain data type
+		// Once added, cannot be removed!
+		template<typename T> inline ComponentManager<T>& Register(std::string name){
+			uint64_t name_hashed = wi::helper::string_hash(name.c_str());
```

Comment:
Thinking about the string hash, I think we might run into hash collision problem, so probably it's better to use strings. That way on hash collision strings can still be compared.